book/src/rfcs/benchmarking.md

```diff
@@ -0,0 +1,266 @@
+them.
+
+## Parameters
+
+It will be of interest parameterizing some benchmarks, for example benchmarking
+verification time but doing it with different circuits sizes, that would show
+how performance scales with the size of the circuit.  The introduction of
+parameters doesn't cause any fundamental change to the benchmarks, but it can
+considerably increase the cost of running them, making them more the kind of
+benchmarks that we may not want to run every single time in CI.
+
+## Benchmark classes
+
+Next the classes of benchmarks to consider, they are different combinations of
+metrics, layers and consumers.
+
+| benchmarks               | repository    | metric   | for external use | for internal use | regression test | noise sensitive  |
+| ------------------------ | ------------- | -------- | ---------------- | ---------------- | --------------- | ---------------- |
+| kimchi time              | proof-systems | time     | yes              | yes              | yes             | yes              |
+| kimchi time flamegraph   | proof-systems | time     | no               | yes              | no              | no               |
+| kimchi instructions      | proof-systems | time     | no               | yes              | yes             | no               |
+| kimchi memory            | proof-systems | memory   | yes              | yes              | yes             | no               |
+| kimchi memory flamegraph | proof-systems | memory   | no               | yes              | no              | no               |
+| snarky time              | mina/snarky*  | time     | yes              | yes              | yes             | yes              |
+| snarky memory            | mina/snarky*  | memory   | yes              | yes              | yes             | no               |
+| pickles time             | mina/pickles* | time     | yes              | yes              | yes             | yes              |
+| pickles memory           | mina/pickles* | memory   | yes              | yes              | yes             | no               |
+
+\* these have to be in Mina now because pickles is in Mina, and while snarky has
+its own repository, using it requires a few things still in Mina, so that we
+might want to consider some changes to make these benchmarks easier and probably
+cheaper.
+
+### Kimchi time
+
+One of the simplest measuring running time, can be a specific case or allow more
+flexibility by introducing parameters like for example different circuit sizes.
+We already have most of the infrastructure for this with
+[criterion](https://crates.io/crates/criterion). We would ideally have a
+dedicated computer. However, considering that there aren't that many PRs in
+proof-systems, the requirements shouldn't be that high.
+
+### Kimchi time flamegraph
+
+This can be considered a variant of the [previous](#kimchi-time). Some more
+research is required but this information ideally can be generated from the
```

Comment:
I think the time for the research is in this document. If it’s too complex to include in the scope of this RFC, then we can remove it for now and address it in another rfc

Reply:
Looked into it, at least for time flamegraphs it shouldn't be too hard, there is [cargo flamegraph](https://github.com/flamegraph-rs/flamegraph) that just wraps the binary and creates the flamegraph, it shouldn't require any change to the code.